src/main/java/com/edmunds/rest/databricks/service/DLTServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.edmunds.rest.databricks.service;
+
+import com.edmunds.rest.databricks.DTO.dlt.PipelineDetailsDTO;
+import com.edmunds.rest.databricks.DTO.dlt.PipelineEventsDTO;
+import com.edmunds.rest.databricks.DTO.dlt.PipelinesDTO;
+import com.edmunds.rest.databricks.DTO.dlt.UpdateDetailsResponseDTO;
+import com.edmunds.rest.databricks.DTO.dlt.UpdateInfoWrapperDTO;
+import com.edmunds.rest.databricks.DatabricksRestException;
+import com.edmunds.rest.databricks.RequestMethod;
+import com.edmunds.rest.databricks.restclient.DatabricksRestClient;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The implementation of DLTService.
+ */
+public class DLTServiceImpl extends DatabricksService implements DLTService {
+
+  public DLTServiceImpl(DatabricksRestClient client) {
+    super(client);
+  }
+
+  @Override
+  public PipelinesDTO listPipelines() throws IOException, DatabricksRestException {
+    byte[] responseBody = client.performQuery(RequestMethod.GET, "/pipelines", new HashMap<>());
+    return this.mapper.readValue(responseBody, PipelinesDTO.class);
+  }
+
+  @Override
+  public PipelineDetailsDTO getPipelineDetails(String pipelineId) throws IOException, DatabricksRestException {
+    String path = String.format("/pipelines/%s", pipelineId);
+    byte[] responseBody = client.performQuery(RequestMethod.GET, path, new HashMap<>());
+    return this.mapper.readValue(responseBody, PipelineDetailsDTO.class);
+  }
+
+  @Override
+  public PipelineDetailsDTO updatePipeline(String pipelineId, Map<String, Object> data)
+          throws IOException, DatabricksRestException {
+    String path = String.format("/pipelines/%s", pipelineId);
+    byte[] responseBody = client.performQuery(RequestMethod.PUT, path, data);
+    return this.mapper.readValue(responseBody, PipelineDetailsDTO.class);
+  }
+
+  @Override
+  public PipelineEventsDTO listPipelineEvents(String pipelineId, String since, String until, String pageToken)
+          throws IOException, DatabricksRestException {
+
+    Map<String, Object> data = new HashMap<>();
+    data.put("max_results", 100);
+    data.put("level", "INFO");
+    if (pageToken != null) {
+      data.put("page_token", pageToken);
+    } else if (since != null && until != null) {
+      String timestampFilterString = String.format("timestamp > '%s' AND timestamp < '%s'", since, until);
```

Comment:
nitpicks:

1. for `since` I think >= would be a better choice of comparison
2. implement timestamp filtering when only one of since/until is provided

Reply:
Fixed